packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -113,30 +114,34 @@ export function addResponderEventSystemEvent(
     document,
   );
   if (passiveBrowserEventsSupported) {
-    addEventCaptureListenerWithPassiveFlag(
+    fbListener = addEventCaptureListenerWithPassiveFlag(
       document,
       topLevelType,
       listener,
       passive,
     );
   } else {
-    addEventCaptureListener(document, topLevelType, listener);
+    fbListener = addEventCaptureListener(document, topLevelType, listener);
   }
-  return listener;
+  return fbListener || listener;
```

Comment:
I'll be honest, this code is a bit confusing. I can trace it out and it looked ok, but I defer to your judgement here.